packages/taiko-client/pkg/chain_iterator/event_iterator/batch_proposed_iterator.go

```diff
@@ -173,16 +173,14 @@ func assembleBatchProposedIteratorCallback(
 		for iterShasta.Next() {
 			event := iterShasta.Event
 
-			proposedEventPayload, err := rpcClient.DecodeProposedEventPayload(&bind.CallOpts{Context: ctx}, event.Data)
+			payload, err := rpcClient.DecodeProposedEventPayload(&bind.CallOpts{Context: ctx}, event.Data)
 			if err != nil {
 				log.Error("Failed to decode proposed event data", "error", err)
 				return err
 			}
-			if proposedEventPayload == nil {
-				return errors.New("decoded proposed event payload is nil")
-			}
```

Comment:
shouldn't we keep this check?

Reply:
Added back: https://github.com/taikoxyz/taiko-mono/pull/20685/commits/1bf722f6601a87fda3c1c76deb37fca6bdc3be05